### PR TITLE
Classify materials

### PR DESCRIPTION
**Pull request by gastonabella**
### Brief Description
Service that classify a group of pockets, given the kg of material.
### Related card or ticket
https://github.com/Pis-moove-it/reciclando-backend/issues/3
### Additional Details
Given an array of pocket ids, kg_trash, kg_recycled_plastic and kg_recycled_glass, it turns each pocket state into `Classified` and distributes each weight proportionally.

### CODE REVIEW

File: app/controllers/classification_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class ClassificationController < AuthenticateController
+  def create
+    pockets = Pocket.where(id: classification_params[:pocket_ids], organization: logged_user.organization)
+    return render_error(1, 'Negative kg of trash') if classification_params[:kg_trash].negative?
```
Comment (fbalarini) [inline app/controllers/classification_controller.rb]: i think we should change this validations to the model

File: app/controllers/classification_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class ClassificationController < AuthenticateController
+  def create
+    pockets = Pocket.where(id: classification_params[:pocket_ids], organization: logged_user.organization)
```
Comment (fbalarini) [inline app/controllers/classification_controller.rb]: Move this to a private method

File: app/controllers/classification_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    return render_error(1, 'Negative kg of trash') if classification_params[:kg_trash].negative?
+    return render_error(1, 'Negative kg of plastic') if classification_params[:kg_plastic].negative?
+    return render_error(1, 'Negative kg of glass') if classification_params[:kg_glass].negative?
+    return render_error(1, 'Unweighed or classified pockets') unless pockets.all?(&:Weighed?)
+    pockets.each do |p|
+      p.classify(pockets.sum(&:weight), classification_params[:kg_trash],
```
Comment (fbalarini) [inline app/controllers/classification_controller.rb]: mmm you are calculating the sum for each pocket, this should be only once

File: app/controllers/classification_controller.rb (hunk 1 above)
Comment (fbalarini) [reply app/controllers/classification_controller.rb]: i saw you already added this numerical validations to the model so you should remove this checks

**Review by fbalarini — DISMISSED** [review]

**Review by accg14 — APPROVED** [review]

**Pull request merged**